modules/ravestate_telegramio/telegram_bot.py

```diff
@@ -57,11 +67,22 @@ def push_telegram_interloc(ctx: ContextWrapper, telegram_node: Node, name: str):
             logger.debug(f"Pushed {telegram_node} to interloc:all")
 
     def make_sure_effective_user_exists(update: Update):
+        """
+        Retrieves scientio Node of User if it exists, otherwise creates it in the scientio session
+        Calls the push_telegram_interloc receptor to push the scientio node into interloc:all
+        Adds the User to the set of active_users
+        """
         if update.effective_user not in active_users:
             # set up scientio
             sess: Session = ravestate_ontology.get_session()
             onto: Ontology = ravestate_ontology.get_ontology()
 
+            tick_interval = 1. / ctx.conf(mod=Context.core_module_name, key=Context.tick_rate_config)
+            while not sess or not onto:
```

Comment:
Damn... 😮 👀 A slightly cleaner solution may be to introduce a `threading.Event` in ontology like `ravestate_ontology.initialized`. You could then use `if ravestate_ontology.initialized.wait(): ...` (See 
https://docs.python.org/2/library/threading.html#threading.Event.wait)

Reply:
Thanks for the suggestion!